streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -70,6 +70,7 @@ public class StreamsProducer {
     private Producer<byte[], byte[]> producer;
     private boolean transactionInFlight = false;
     private boolean transactionInitialized = false;
+    private boolean resetDisabled = false;
```

Comment:
It seems to be a leaking abstraction to put this flag into `StreamsProducer`?

Would it be better to add a `boolean isClosed` flag to `ActiveTaskCreator` instead?

Reply:
Let's hear from Sophie.

To me, "disabling the reset" is not really a feature of the producer, but a feature of the `TaskManager` (and in transition the `ActiveTaskCreator` as a helper class used by `TaskManager`). -- Given that only the `TaskManager` decides if it wants to skip the re-initialization or not (and the `TaskManager` does already skip the call by checking `streamsProducer.isResetDisabled()`), it's unclear to my why we would push this flag inside `StreamsProducer`. It seems to be an unnecessary "de-tour", and boilerplate code.

Reply:
yeah I think I agree with Matthias here. let's keep this encapsulated in the activeTaskCreator